packages/core/src/tracing/backend_mappers/mapper.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+/**
+ * @typedef {Object<string, string>} FieldMapping
+ * @typedef {Object<string, FieldMapping>} FieldMappings
+ */
+
+/**
+ * Field mappings for different span types.
+ *
+ * This FieldMappings defines how internal span fields are mapped to backend fields
+ * for various span types (e.g., dynamodb, redis).
+ *
+ * As new span types needs to add, simply add their respective mappings
+ * following the same format (e.g., 'internal-field': 'backend-field').
+ *
+ * @type {FieldMappings}
+ */
+const fieldMappings = {
+  dynamodb: {
+    /// Maps internal field 'operation' to backend field 'op'
+    operation: 'op'
+  },
+  redis: {
+    operation: 'command'
+  }
+};
+
+/**
+ * Transforms span data fields to match the backend format.
+ *
+ * @param {import('../../core').InstanaBaseSpan} span
+ * @returns {import('../../core').InstanaBaseSpan} The transformed span.
+ */
+module.exports.transform = span => {
+  const spanName = span.n;
+  if (!fieldMappings[spanName]) return span;
+
+  const mappings = fieldMappings[spanName];
+  if (span.data[spanName]) {
+    Object.entries(mappings).forEach(([internalField, backendField]) => {
+      if (span.data[spanName][internalField]) {
```

Comment:
issue: IMO its more efficient to iterate over the span.data keys. By that we iterate once in best case. Most of our spans only have one key.

Reply:
a good point! 💵